src/LinkedCells/LinkedCells.cpp

```diff
@@ -0,0 +1,157 @@
+//
+// Created by jv_fedora on 19.11.25.
+//
+
+#include "LinkedCells.h"
+
+#include <spdlog/spdlog.h>
+
+LinkedCells::LinkedCells(std::vector<Particle> &particles, const Vector3 domain, const double cutoff,
+                         std::array<BorderType, 6> border)
+    : particles(particles), domain_size(domain) {
+  // calculate number of cells - should always be at least 1
+  numCellsX = std::max(1, static_cast<int>(domain_size[0] / cutoff));
+  numCellsY = std::max(1, static_cast<int>(domain_size[1] / cutoff));
+  numCellsZ = std::max(1, static_cast<int>(domain_size[2] / cutoff));
+
+  // calculate cell-size dimensions
+  cellSizeX = domain_size[0] / numCellsX;
+  cellSizeY = domain_size[1] / numCellsY;
+  cellSizeZ = domain_size[2] / numCellsZ;
+  cell_size = {cellSizeX, cellSizeY, cellSizeZ};
+
+  // Reserve space for ghost cells
+  numCellsX += 2;
+  numCellsY += 2;
+  numCellsZ += 2;
+
+  cells.resize(numCellsX * numCellsY * numCellsZ);
+
+  for (int i = 0; i < cells.size(); i++) {
+    // check if cell should be ghost cell
+    auto [x, y, z] = index1dToIndex3d(i);
+    if (x == 0 || y == 0 || z == 0 || x == numCellsX - 1 || y == numCellsY - 1 || z == numCellsZ - 1) {
+      cells[i].cell_type = CellType::GHOST;
+    }
+    // check if cell should be a border cell
+    else {
+      setNeighbourCells(i);
+      if (x == 1 || y == 1 || z == 1 || x == numCellsX - 2 || y == numCellsY - 2 || z == numCellsZ - 2) {
+        cells[i].cell_type = CellType::BORDER;
+        // Set Border Types
+        if (x == 1) {
```

Comment:
Can one refactor this somehow, maybe with  `std::array` and a pattern? These many if-statements look "vulnerable"